### PR TITLE
Vend connector for access and erasure

### PR DESCRIPTION
**Pull request by Kelsey-Ethyca**
Closes #[1869](https://github.com/ethyca/fides/issues/1869)

### Code Changes
NEW SaaS Connector Vend for access and erasure 

### Steps to Confirm
* [x] Add/configure the Vend connector in UI
* [x] Add customer data to Vend using Vend UI (no api to add sales) 
* [x] Create an Access DSR with the new vend customer email 
* [x] Verify email with customer data is correct 
* [x] Verify Vend config and icon UI looks good 
* [x] Create an Delete DSR with the new vend customer email 
* [x] Verify the customer data is now masked via Vend api

### Pre-Merge Checklist
* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [x] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### SaaS Connector Checklist 
* [x] [Spec doc link](https://ethyca.atlassian.net/wiki/spaces/EN/pages/2576023553/Vend+Connector+Specification)
* [x] Vault Creds
* [x] Postman Collection
* [x] SaaS Config And Dataset 
* [x] Icon

### CODE REVIEW

File: data/saas/dataset/vend_dataset.yml [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+          - name: mobile
+            data_categories: [user.contact.phone_number]
+            fidesops_meta:
+              data_type: string
+          - name: fax
+            data_categories: [user.contact]
+            fidesops_meta:
+              data_type: string
+          - name: twitter
+            data_categories: [user]
```
Comment (Kelsey-Ethyca) [inline data/saas/dataset/vend_dataset.yml:81]: Should this data category be something more that user?

File: data/saas/dataset/vend_dataset.yml [hunk 2]
```diff
@@ -0,0 +1,453 @@
+dataset:
+  - fides_key: <instance_fides_key>
+    name: Vend Dataset
+    description: A sample dataset representing the Vend connector for Fides
+    collections:
+      - name: customer
+        fields:
+          - name: name
+            data_categories: [user.name]
+            fidesops_meta:
```
Comment (Kelsey-Ethyca) [inline data/saas/dataset/vend_dataset.yml:10]: should this be fides_meta?

**Comment by Kelsey-Ethyca** [comment]
Testing passed 
localhost_3000_datastore-connection_new_step=1
localhost_3000_datastore-connection_new_step=1 (1)
localhost_3000_privacy-requests

File: data/saas/dataset/vend_dataset.yml (hunk 2 above)
Comment (Kelsey-Ethyca) [reply data/saas/dataset/vend_dataset.yml:10]: all connectors are fidesops_meta so we are leaving as is

File: data/saas/dataset/vend_dataset.yml (hunk 1 above)
Comment (Kelsey-Ethyca) [reply data/saas/dataset/vend_dataset.yml:81]: this is all we need

**Review by galvana — APPROVED** [review]
The tests all passed right away which was nice! I just made some minor changes to the dataset and icon dimensions.

**Pull request merged**